authorizerd.go

```diff
@@ -201,18 +203,18 @@ func (a *authorizer) initVerifiers() error {
 		verifiers = append(verifiers, rcVerifier)
 	}
 
-	if len(a.atpParams) > 0 {
+	if a.atpParam.verifyOAuth2AccessToken {
 		atVerifier := func(r *http.Request, act, res string) error {
 			tokenString, err := request.AuthorizationHeaderExtractor.ExtractToken(r)
 			if err != nil {
 				return err
 			}
-			if r.TLS != nil {
+			if r.TLS != nil && len(r.TLS.PeerCertificates) != 0 {
```

Comment:
Changed it because even if the r.TLS is not nil, there could be no certificates.